.github/actions/azure/action.yml

```diff
@@ -0,0 +1,45 @@
+# This GitHub Action is designed to execute Azure CLI for infrastructure maintenance
+
+name: 'Deploy Azure WebApp'
+description: 'This custom GitHub Actions module will execute AZ CLI for infrastructure maintenance'
+
+# Define the inputs required for this action.
+inputs:
+  credentials:
+    description: 'Azure subscription and resource group service principal'
+    required: true
+  subscription:
+    description: 'Azure subscription ID'
+    required: true
+  group:
+    description: 'Azure resource group'
+    required: true
+  acr:
+    description: 'Azure subscription and resource group service principal'
+    required: true
+  days:
+    description: 'Retention days for ACR artifacts purge i.e images older than x days will be removed.'
+    required: false
+    default: '30'
+
+# Define the steps to run this action.
+runs:
+  using: 'composite'
+  steps:
+    # Step 1: Check out the repository.
+    - name: Repository 🗃️
+      uses: actions/checkout@v4
+
+
+    # Step 2: Log in to Azure.
+    - name: Azure 🔐
+      uses: azure/login@v2
+      with:
+        creds: ${{ inputs.credentials }}
+
+    # Step 3: Purge artifacts
```

Comment:
do we care about keeping the 'latest' image while purging everything else (arguably this isn't on prod yet so unlikely latest is over 30 days old)

Reply:
Good question, have added an optional `keep` attribute which default value of `3`